### ISSUE TITLE
Optimize canvas rendering 

### ISSUE THREAD

**Issue by ykevu** [vanvalenlab/deepcell-label#412]
For larger images with many cells, canvas rendering updates can become quite slow, largely due to the way overlaps are handled. We can either migrate to a higher level library/framework such as deck.gl or utilize more potential optimizations in the canvas rendering kernels. In the short term, one method of handling this would be using a modified label format where overlap values are strictly negative, and enforcing that a cell id must map to that value id as well, for instance:
```
[{cell: 0, value: 0, t: 0, c: 0}, 
{cell: 1, value: 1, t: 0, c: 0}, 
{cell: 0, value: -1, t: 0, c: 0}, 
{cell: 1, value: -1, t: 0, c: 0}]
```
Then, we can store a list of overlaps such as ``` overlaps=[[0, 1]] ```, where the _i_ th index in ```overlaps``` corresponds to value ```0 - i - 1```, and the labeled and cell types canvases can separately handle positive values (just index into the colormap) and negative values (index into overlaps to obtain the list of overlapping cells present, then index into colormap for each cell).

Currently, we _always_  iterate through the number of cells in the canvas for each pixel, giving us _O(n)_ where _n_ is the number of cells. In the worst case with this new optimization, every pixel contains every cell, in which case we are still _O(n)_. However, we would expect that we are not handling images of this kind, but instead images where <10% of the pixels are overlapping, so our average time complexity is significantly better.

**Comment by ykevu** [comment]
Though this strategy could work if we are OK with overhauling the data format, a short term solution for cell types in particular is offered in #437.

**Comment by ykevu** [comment]
An alternative: potential cell matrix format, with two arrays:

_cellMapping_: `[[0], [1], [2, 3]]`

_numCellsArray_: `[1, 1, 2]`

The _i_ th array in _cellMapping_ represents a list of the cells that map to value _i_
The _j_ th element of _numCellsArray_ represents the number of cells that map to value _j_

We can then have a kernel something like:

```
value = labelArray[y][x]
numCells = numCellsArray[value]
for (let i = 0; i < numCells; i++) {
    cell = cellMapping[numCells[i]]
    color = colorMap[cell]
    appendColorToOverlap(color)
}
```

**Comment by ykevu** [comment]
Implemented with #502
